### PR TITLE
"IRC Bot" support.

### PR DESCRIPTION
**Pull request by AndrewMontagne**
No changelog as this is all backend. This does not include the script that is called.

image
image

### CODE REVIEW

File: code/modules/admin/verbs/adminhelp.dm [hunk 1]
```diff
@@ -568,8 +568,8 @@ GLOBAL_DATUM_INIT(ahelp_tickets, /datum/admin_help_tickets, new)
 	var/list/adm = get_admin_counts(requiredflags)
 	var/list/activemins = adm["present"]
 	. = activemins.len
-	if(. <= 0)
-		var/final = ""
+	if (. <= 0)
```
Comment (SamHPurp) [inline code/modules/admin/verbs/adminhelp.dm]: reee

File: code/modules/server_tools/server_tools.dm [hunk 2]
```diff
@@ -2,17 +2,28 @@ GLOBAL_VAR_INIT(reboot_mode, REBOOT_MODE_NORMAL)	//if the world should request t
...
 	if(world.RunningService())
 		return world.params[SERVICE_VERSION_PARAM]
 
 /world/proc/ExportService(command)
-	return RunningService() && shell("python code/modules/server_tools/nudge.py \"[command]\"") == 0
+	if (world.RunningService())
+		var/script = CONFIG_GET(string/service_command)
+		var/status = shell("[script] [command]")
+		log_world("SERVICE COMMAND: [script] [command]")
+		if (status == null)
```
Comment (SamHPurp) [inline code/modules/server_tools/server_tools.dm]: You are putting spaces after `if`s, different to the rest of the codebase yo.

Also, can just `!status`?

File: code/modules/server_tools/server_tools.dm [hunk 3]
```diff
@@ -2,17 +2,28 @@ GLOBAL_VAR_INIT(reboot_mode, REBOOT_MODE_NORMAL)	//if the world should request t
...
-	return params[SERVICE_WORLD_PARAM]
+	return CONFIG_GET(string/service_command) != null
 
 /proc/ServiceVersion()
 	if(world.RunningService())
 		return world.params[SERVICE_VERSION_PARAM]
 
 /world/proc/ExportService(command)
-	return RunningService() && shell("python code/modules/server_tools/nudge.py \"[command]\"") == 0
+	if (world.RunningService())
```
Comment (SamHPurp) [inline code/modules/server_tools/server_tools.dm]: You are putting spaces after `if`s, different to the rest of the codebase yo.

File: code/modules/server_tools/server_tools.dm (hunk 2 above)
Comment (AndrewMontagne) [reply code/modules/server_tools/server_tools.dm]: Need to explicitly check it is null and not 0

**Pull request merged**